python/cudf/cudf/core/column/column.py

```diff
@@ -1809,59 +1818,20 @@ def as_column(
         return col
 
     elif isinstance(arbitrary, (pa.Array, pa.ChunkedArray)):
-        if pa.types.is_large_string(arbitrary.type):
-            # Pandas-2.2+: Pandas defaults to `large_string` type
-            # instead of `string` without data-introspection.
-            # Temporary workaround until cudf has native
-            # support for `LARGE_STRING` i.e., 64 bit offsets
-            arbitrary = arbitrary.cast(pa.string())
-
-        if pa.types.is_float16(arbitrary.type):
-            raise NotImplementedError(
-                "Type casting from `float16` to `float32` is not "
-                "yet supported in pyarrow, see: "
-                "https://github.com/apache/arrow/issues/20213"
-            )
-        elif (
-            pa.types.is_timestamp(arbitrary.type)
-            and arbitrary.type.tz is not None
-        ):
-            raise NotImplementedError(
-                "cuDF does not yet support timezone-aware datetimes"
-            )
-        elif (nan_as_null is None or nan_as_null) and pa.types.is_floating(
+        if (nan_as_null is None or nan_as_null) and pa.types.is_floating(
             arbitrary.type
         ):
             arbitrary = pc.if_else(
                 pc.is_nan(arbitrary),
                 pa.nulls(len(arbitrary), type=arbitrary.type),
                 arbitrary,
             )
+        elif dtype is None and pa.types.is_null(arbitrary.type):
+            # default "empty" type
+            dtype = "str"
         col = ColumnBase.from_arrow(arbitrary)
 
-        if isinstance(arbitrary, pa.NullArray):
-            if dtype is not None:
-                # Cast the column to the `dtype` if specified.
-                new_dtype = dtype
-            elif len(arbitrary) == 0:
-                # If the column is empty, it has to be
-                # a `str` dtype.
-                new_dtype = cudf.dtype("str")
-            else:
-                # If the null column is not empty, it has to
-                # be of `object` dtype.
-                new_dtype = cudf.dtype(arbitrary.type.to_pandas_dtype())
-
-            if cudf.get_option(
-                "mode.pandas_compatible"
-            ) and new_dtype == cudf.dtype("O"):
-                # We internally raise if we do `astype("object")`, hence
-                # need to cast to `str` since this is safe to do so because
-                # it is a null-array.
-                new_dtype = "str"
-
-            col = col.astype(new_dtype)
-        elif dtype is not None:
```

Comment:
Why don't we need to add handling for this into `from_arrow`? The other removals here all have a corresponding branch in `from_arrow`. Is this now being handled by the new three lines you added above that are doing `pa.types.is_null`?

Reply:
> Why don't we need to add handling for this into from_arrow?

I had kept this here since `from_arrow` doesn't do any explicit casting with `dtype`. But, are `pyarrow.null` type arrays ever expected to be represented as is in cudf? Otherwise, it would make sense to convert null to str in `from_arrow`

> Is this now being handled by the new three lines you added above that are doing pa.types.is_null?

Yes, I think this casting was here when the default "empty" type was float pre-pandas 2.0 support and we need this to be string sometimes. Now we can collapse all these conditions to always cast to string

Reply:
I think pyarrow.null has like one possible code path in all of the cudf test suite where it gets used :sweat_smile: so I think you can safely ignore it as long as tests are currently passing. Eventually we'll need to deal with this more thoroughly in #12477